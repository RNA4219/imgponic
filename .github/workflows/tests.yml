name: ci

on:
  pull_request:
  push:
    branches:
      - main

jobs:
  frontend:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - name: Cache npm
        uses: actions/cache@v4
        with:
          path: ~/.npm
          key: ${{ runner.os }}-npm-${{ hashFiles('package-lock.json', 'package.json') }}
          restore-keys: |
            ${{ runner.os }}-npm-

      - name: Setup Node.js
        uses: actions/setup-node@v4
        with:
          node-version: '20'

      - name: Install dependencies
        run: npm install

      - name: Build frontend
        run: npm run build

      - name: Run frontend tests
        run: npm run test

  rust:
    runs-on: ubuntu-latest
    needs: frontend
    steps:
      - uses: actions/checkout@v4

      - name: Setup Rust toolchain
        uses: actions-rs/toolchain@v1
        with:
          toolchain: stable
          profile: minimal
          override: true

      - name: Install GTK dependencies
        run: sudo apt-get update && sudo apt-get install -y pkg-config libgtk-3-dev libglib2.0-dev libgdk-pixbuf-2.0-dev libpango1.0-dev libsoup2.4-dev libwebkit2gtk-4.1-dev libappindicator3-dev

      - name: Check formatting
        run: cargo fmt -- --check

      - name: Lint with clippy
        run: cargo clippy -- -D warnings

      - name: Run tests
        run: cargo test

  completion:
    runs-on: ubuntu-latest
    needs:
      - frontend
      - rust
    if: ${{ always() }}
    steps:
      - name: Report status
        run: |
          if [[ "${{ needs.frontend.result }}" != 'success' || "${{ needs.rust.result }}" != 'success' ]]; then
            echo 'One or more jobs failed'
            exit 1
          fi
          echo 'All checks succeeded'
