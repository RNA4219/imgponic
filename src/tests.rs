pub use super::_compose_prompt;
