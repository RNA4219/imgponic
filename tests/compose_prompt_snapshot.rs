use std::fs;

use promptforge::tests::_compose_prompt;

#[test]
fn compose_prompt_snapshot() {
    let temp = tempfile::tempdir().expect("failed to create temp dir");
    let base = temp.path();

    let recipes_dir = base.join("recipes");
    let fragments_dir = base.join("fragments/system");
    fs::create_dir_all(&recipes_dir).expect("failed to create recipes dir");
    fs::create_dir_all(&fragments_dir).expect("failed to create fragments dir");

    fs::write(
        recipes_dir.join("demo.yaml"),
        r"profile: llama3
fragments:
  - system.prompt
params:
  app_name: PromptForge
  current_date: 2024-01-01
  user_input: Please summarize the document.
",
    )
    .expect("failed to write recipe");

    fs::write(
        fragments_dir.join("prompt.yaml"),
        r"id: system.prompt
kind: system
content: |
  System instructions for {{app_name}}.
  Current date: {{current_date}}
",
    )
    .expect("failed to write fragment");

    let recipe_path = recipes_dir.join("demo.yaml");

    let result = _compose_prompt(recipe_path.to_str().unwrap(), None).expect("compose prompt");

    insta::assert_snapshot!(result.final_prompt);
}
