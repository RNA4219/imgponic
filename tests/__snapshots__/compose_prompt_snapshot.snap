---
source: tests/compose_prompt_snapshot.rs
expression: result.final_prompt
---
System instructions for PromptForge.
Current date: 2024-01-01

---
USER_INPUT (verbatim):
```text
Please summarize the document.
```
